contribs/gmf/src/services/authenticationservice.js

```diff
@@ -0,0 +1,174 @@
+goog.provide('gmf.AuthenticationEventType');
+
+goog.require('gmf');
+goog.require('goog.events.EventTarget');
+goog.require('goog.uri.utils');
+
+
+/**
+ * @enum {string}
+ */
+gmf.AuthenticationEventType = {
+  LOAD: 'load'
+};
+
+
+/**
+ * @typedef {{
+ *     logged: boolean
+ * }}
+ */
+gmf.AuthenticationIsLoggedInResponse;
+
+
+/**
+ * @typedef {{
+ *     status: boolean
+ * }}
+ */
+gmf.AuthenticationLoginResponse;
+
+
+/**
+ * @typedef {{
+ *     status: boolean
+ * }}
+ */
+gmf.AuthenticationResetPasswordResponse;
+
+
+
+/**
+ * @constructor
+ * @extends {goog.events.EventTarget}
+ * @param {angular.$http} $http Angular http service.
+ * @param {string} authenticationBaseUrl URL to "authentication" web service.
+ * @ngInject
+ */
+gmf.Authentication = function($http, authenticationBaseUrl) {
+
+  goog.base(this);
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.baseUrl_ = authenticationBaseUrl;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.isLoggedInRouteSuffix_ = 'isloggedin';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.loginRouteSuffix_ = 'login';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.resetPasswordRouteSuffix_ = 'loginresetpassword';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  //this.logoutRouteSuffix_ = 'logout';
+
+  /**
+   * @type {?boolean}
+   * @export
+   */
+  this.isLoggedIn = null;
+
+  this.load_();
+};
+goog.inherits(gmf.Authentication, goog.events.EventTarget);
+
+
+/**
+ * Load the authentication service, which sends an asynch request to
+ * determine whether the user is currently connected or not.
+ * @private
+ */
+gmf.Authentication.prototype.load_ = function() {
+  var isLoggedInUrl = goog.uri.utils.appendPath(
+      this.baseUrl_, this.isLoggedInRouteSuffix_);
+  this.http_.get(isLoggedInUrl, {
+    cache: false
+  }).then(goog.bind(
+      /**
+       * @param {angular.$http.Response} resp Ajax response.
+       * @return {gmf.AuthenticationIsLoggedInResponse}
+       */
+      function(resp) {
+        var data = /** @type {gmf.AuthenticationIsLoggedInResponse} */ (
+            resp.data);
+        this.isLoggedIn = data.logged;
+        this.dispatchEvent(gmf.AuthenticationEventType.LOAD);
+        return data;
+      }, this));
+};
+
+
+/**
+ * @param {string} login
+ * @param {string} pwd
+ * @param {Function} callback
+ * @export
+ */
+gmf.Authentication.prototype.login = function(login, pwd, callback) {
+
+  var url = goog.uri.utils.appendPath(this.baseUrl_, this.loginRouteSuffix_);
+
+  // FIXME - use post
+  this.http_.get(url, {
+    login: login,
+    password: pwd
+  })
+    .then(goog.bind(function(response) {
+        var responseData = /** @type {gmf.AuthenticationLoginResponse} */ (
+            response.data);
+        if (responseData.status) {
+          this.isLoggedIn = true;
+        }
+        callback(responseData);
```

Comment:
No this is not the way it should be done.
Look at the themservice, it should return $promises, then you call your callback from where you call the method (don't pass a callback to the method).
https://github.com/fgravin/ngeo/blob/9a95af08e373d1928394bf8cbec03ae0c3499739/contribs/gmf/src/services/themesservice.js#L103

Watch doc https://docs.angularjs.org/api/ng/service/$q